apps/pie-storybook/stories/testing/pie-button.test.stories.ts

```diff
@@ -0,0 +1,477 @@
+                    <span>Name: </span>
+                    <strong><span aria-label="required">*</span></strong>
+                </label>
+                <input type="text" id="name" data-test-id="name" name="username" required />
+            </p>
+            <p>
+                <label for="mail">
+                    <span>E-mail: </span>
+                    <strong><span aria-label="required">*</span></strong>
+                </label>
+                <input type="email" data-test-id="usermail" id="mail" name="usermail" required />
+            </p>
+            <p>
+                <label for="pwd">
+                    <span>Password: </span>
+                    <strong><span aria-label="required">*</span></strong>
+                </label>
+                <input type="password" data-test-id="password" id="pwd" name="password" required />
+            </p>
+        </section>
+        <section>
+            <h2>Payment information</h2>
+            <p>
+                <label for="card">
+                    <span>Card type:</span>
+                </label>
+                <select data-test-id="usercard" id="card" name="usercard">
+                    <option value="visa">Visa</option>
+                    <option value="mastercard">Mastercard</option>
+                    <option value="amex">American Express</option>
+                </select>
+            </p>
+            <p>
+                <label for="number">
+                    <span>Card number:</span>
+                    <strong><span aria-label="required">*</span></strong>
+                </label>
+                <input type="tel" data-test-id="card-number" id="number" name="cardnumber" required />
+            </p>
+            <p>
+                <label for="expiration">
+                    <span>Expiration date:</span>
+                    <strong><span aria-label="required">*</span></strong>
+                </label>
+                <input type="text" data-test-id="card-expiration" id="expiration" required placeholder="MM/YY"
+                    pattern="^(0[1-9]|1[0-2])\/([0-9]{2})$" />
+            </p>
+        </section>
+        <section style="display: flex; gap: var(--dt-spacing-a); justify-content: flex-end; flex-wrap: wrap; margin-top: var(--dt-spacing-b);">
+            ${showNativeResetButton ? html`<button data-test-id="button--reset" id="resetNativeButton" type="reset">Reset</button>` : Template({
+    ...props,
+    variant: 'secondary',
+    slot: 'Reset',
+    type: 'reset',
+})}
+            ${showSubmitButton ? Template({
+    ...props,
+    variant: 'primary',
+    slot: 'Submit payment',
+    type: 'submit',
+}) : nothing}
+        </section>
+    </form>
+    <script>
+        // var is used to prevent storybook from erroring when the script is re-run
+        var form = document.querySelector('#testForm');
+        var formLog = document.querySelector('#formLog');
+
+        form.addEventListener('submit', (e) => {
```

Comment:
Please can we consider using the method shown in `TextInput` for event handling in Story templates? [example](https://github.com/justeattakeaway/pie/blob/7ba32a8a01137d87537342c9e35f10a49256a813/apps/pie-storybook/stories/pie-text-input.stories.ts#L257)

I believe you can then use `@submit` on the form element and avoid having to add this script tag in the HTML (I think it should work here)